src/DSharp.Compiler/Compiler/ExpressionBuilder.cs

```diff
@@ -1825,6 +1820,10 @@ private Expression ProcessOpenParenExpressionNode(BinaryExpressionNode node)
                 }
 
                 methodExpression = new MethodExpression(memberExpression.ObjectReference, method);
+
+                // iterate over the parameters. If the parameter is generic, then get args[i].EvaluatedType
```

Comment:
this is what needs to be done in order to support `typeof(T)` and `T` as return type

Reply:
We then also need to include those type parameters in the method parameters when invoked by callers, same with if a class is constructed, we need a mechanism to pass those generic parameters in, as the type check is a runtime check